libdevcore/Worker.h

```diff
@@ -29,74 +29,60 @@
 namespace dev
 {
 
-enum class IfRunning
-{
-	Fail,
-	Join,
-	Detach
-};
-
 enum class WorkerState
 {
-	Starting,
-	Started,
-	Stopping,
-	Stopped,
```

Comment:
This makes sense and it's a reversion to the prior logic with support for terminate().

So the requested state has been separated from the actual state. In total, removing about two LOC and adding another atomic. Are two LOC worth the additional state? Why introduce additional state and complexity when it isn't necessary?